dask/array/tests/test_masked.py

```diff
@@ -28,6 +29,23 @@ def test_from_array_masked_array():
     assert_eq(dm, m)
 
 
+def test_copy_deepcopy():
+    t = np.ma.masked_array([1, 2], mask=[0, 1])
+    x = da.from_array(t, chunks=t.shape, asarray=False)
+    #x = da.arange(5, chunks=(2,))
+    y = x.copy()
+    memo = {}
+    y2 = deepcopy(x, memo=memo)
+
+    xx = da.ma.masked_where([False, True], [1,2])
+    assert_eq(x, xx)
+
+    assert_eq(y, t)
+    assert isinstance(y.compute(), np.ma.masked_array)
```

Comment:
Are type checks not already part of `assert_eq`?  If not, then maybe we should add that generally.  It's nice to apply checks to `assert_eq` because then they get applied everywhere.